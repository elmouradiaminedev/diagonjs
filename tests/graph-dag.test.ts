import { expect, describe, it, beforeAll } from "vitest";
import Diagon, { DiagonType } from "src";

let diagon: DiagonType;

beforeAll(async () => {
  diagon = await Diagon.init();
});

describe("GraphDAG expression translation", () => {
  it("should translate clothes DAG", () => {
    expect(
      diagon.translate.graphDAG(
        "socks -> shoes\nunderwear -> shoes\nunderwear -> pants\npants -> shoes\npants -> belt\nbelt -> jacket\nshirt -> belt\nshirt -> tie\ntie -> jacket\n",
      ),
    ).toMatchSnapshot();
  });

  it("should translate chromium DAG", () => {
    expect(
      diagon.translate.graphDAG(
        "chrome -> content\nchrome -> blink\nchrome -> base\n\ncontent -> blink\ncontent -> net\ncontent -> base\n\nblink -> v8\nblink -> CC\nblink -> WTF\nblink -> skia\nblink -> base\nblink -> net\n\nweblayer -> content\nweblayer -> chrome\nweblayer -> base\n\nnet -> base\nWTF -> base\n",
      ),
    ).toMatchSnapshot();
  });

  it("should translate compilation DAG", () => {
    expect(
      diagon.translate.graphDAG(
        "random -> pool_urbg\nrandom -> nonsecure_base\nrandom -> seed_sequence\nrandom -> distribution\n\nnonsecure_base -> pool_urbg\nnonsecure_base -> salted_seed_seq\n\nseed_sequence -> pool_urbg\nseed_sequence -> salted_seed_seq\nseed_sequence -> seed_material\n\ndistribution -> strings\n\npool_urbg -> seed_material\n\nsalted_seed_seq -> seed_material\n\nseed_material -> strings\n",
      ),
    ).toMatchSnapshot();
  });
});
