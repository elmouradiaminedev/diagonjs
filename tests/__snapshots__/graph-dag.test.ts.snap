// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GraphDAG expression translation > should translate chromium DAG 1`] = `
"┌────────┐                     
│weblayer│                     
└┬─┬─┬───┘                     
 │ │┌▽────────────────────────┐
 │ ││chrome                   │
 │ │└┬─────┬─────────────────┬┘
 │┌▽─▽────┐│                 │ 
 ││content││                 │ 
 │└┬─┬─┬──┘│                 │ 
 │ │ │┌▽───▽──────────────┐  │ 
 │ │ ││blink              │  │ 
 │ │ │└┬──┬───┬─┬───┬───┬─┘  │ 
 │ │┌▽─▽┐┌▽──┐│┌▽─┐┌▽─┐┌▽───┐│ 
 │ ││net││WTF│││v8││CC││skia││ 
 │ │└┬──┘└┬──┘│└──┘└──┘└────┘│ 
┌▽─▽─▽────▽───▽──────────────▽┐
│base                         │
└─────────────────────────────┘
"
`;

exports[`GraphDAG expression translation > should translate clothes DAG 1`] = `
"┌─────┐┌─────────┐┌─────┐     
│socks││underwear││shirt│     
└┬────┘└┬─┬──────┘└┬─┬──┘     
 │      │┌▽─────┐  │┌▽───────┐
 │      ││pants │  ││tie     │
 │      │└┬──┬──┘  │└┬───────┘
┌▽──────▽─▽┐┌▽─────▽┐│        
│shoes     ││belt   ││        
└──────────┘└┬──────┘│        
┌────────────▽───────▽┐       
│jacket               │       
└─────────────────────┘       
"
`;

exports[`GraphDAG expression translation > should translate compilation DAG 1`] = `
"┌───────────────────────────────┐             
│random                         │             
└┬─────────────┬─────────────┬─┬┘             
┌▽───────────┐┌▽────────────┐│┌▽─────────────┐
│distribution││seed_sequence│││nonsecure_base│
└┬───────────┘└┬───┬───────┬┘│└┬┬────────────┘
 │             │  ┌│───────│─│─│┘             
 │ ┌───────────┘  ││       │ │┌┘              
 │ │┌─────────────▽▽┐┌─────▽─▽▽┐              
 │ ││salted_seed_seq││pool_urbg│              
 │ │└┬──────────────┘└┬────────┘              
 │┌▽─▽────────────────▽┐                      
 ││seed_material       │                      
 │└┬───────────────────┘                      
┌▽─▽────┐                                     
│strings│                                     
└───────┘                                     
"
`;
